server/controllers/work/editWork.js

```diff
@@ -0,0 +1,39 @@
+const { Work } = require('../../database/models');
+const { customError } = require('../errors');
+const { workEditValidation } = require('../../utils/validation');
+const { uploadImage, deleteImage } = require('../../utils');
+
+const editWork = async (req, res, next) => {
+  try {
+    const { id } = await workEditValidation.validateAsync(req.body);
+    const { title, content } = req.body;
+    const { providerID } = req;
+    const work = await Work.findByPk(id);
+    if (!work) {
+      throw customError('work does not exist', 400);
+    }
+    if (work.userId !== +providerID) {
+      throw customError('Unauthorized', 401);
+    }
+    let imgUrl = '';
+    if (req.files) {
+      const { image } = req.files;
+      imgUrl = await uploadImage(image, work.userId);
+      deleteImage(work.image);
```

Comment:
Also the delete image take the key, which is part of the url, But not the whole url
So when the url is 
`https://herfa.s3.eu-west-2.amazonaws.com/2/f7454dc0-cf6a-11ec-a5f1-61119f11f019.png`
the key woukd be 
`2/f7454dc0-cf6a-11ec-a5f1-61119f11f019.png`
So try to split after `url.com`